opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -25,55 +25,38 @@
 from opentelemetry.util._importlib_metadata import entry_points
 
 logger = logging.getLogger(__name__)
-_RUNTIME_CONTEXT = None  # type: typing.Optional[_RuntimeContext]
-_RUNTIME_CONTEXT_LOCK = threading.Lock()
 
-_F = typing.TypeVar("_F", bound=typing.Callable[..., typing.Any])
-
-
-def _load_runtime_context(func: _F) -> _F:
-    """A decorator used to initialize the global RuntimeContext
+def _load_runtime_context() -> typing.Optional[_RuntimeContext]:
+    """Initialize the RuntimeContext
 
     Returns:
-        A wrapper of the decorated method.
+        An instance of RuntimeContext.
     """
 
-    @wraps(func)  # type: ignore[misc]
-    def wrapper(
-        *args: typing.Tuple[typing.Any, typing.Any],
-        **kwargs: typing.Dict[typing.Any, typing.Any],
-    ) -> typing.Optional[typing.Any]:
-        global _RUNTIME_CONTEXT  # pylint: disable=global-statement
-
-        with _RUNTIME_CONTEXT_LOCK:
-            if _RUNTIME_CONTEXT is None:
-                # FIXME use a better implementation of a configuration manager
-                # to avoid having to get configuration values straight from
-                # environment variables
-                default_context = "contextvars_context"
-
-                configured_context = environ.get(
-                    OTEL_PYTHON_CONTEXT, default_context
-                )  # type: str
-                try:
-
-                    _RUNTIME_CONTEXT = next(  # type: ignore
-                        iter(  # type: ignore
-                            entry_points(  # type: ignore
-                                group="opentelemetry_context",
-                                name=configured_context,
-                            )
-                        )
-                    ).load()()
-
-                except Exception:  # pylint: disable=broad-except
-                    logger.exception(
-                        "Failed to load context: %s", configured_context
-                    )
-        return func(*args, **kwargs)  # type: ignore[misc]
-
-    return typing.cast(_F, wrapper)  # type: ignore[misc]
+    # FIXME use a better implementation of a configuration manager
+    # to avoid having to get configuration values straight from
+    # environment variables
+    default_context = "contextvars_context"
+
+    configured_context = environ.get(
+        OTEL_PYTHON_CONTEXT, default_context
+    )  # type: str
+
+    try:
+        return next(  # type: ignore
+            iter(  # type: ignore
+                entry_points(  # type: ignore
+                    group="opentelemetry_context",
+                    name=configured_context,
+                )
+            )
+        ).load()()
+    except Exception:  # pylint: disable=broad-except
```

Comment:
Looks like the mypy check is failing because it wants to see a return statement for the case where an exception was thrown.

Reply:
Would it be better to just throw the exception instead of catch it for **failfast**? Because the exception only happens when environment variable `OTEL_PYTHON_CONTEXT` set to unsupported value. It's also fine to **failover** to the `default_context`, however, it requires extra codes to load the `default_context` on exception catching.

In short, I prefer **failfast**, just remove `try/except`.

Reply:
Fixed